crates/oxc_linter/src/rules/node/no_new_require.rs

```diff
@@ -0,0 +1,73 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_new_require(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unexpected use of new with require")
+        .with_label(span)
+        .with_help("Initialise the constructor separate from the import statement")
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNewRequire;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Warn about calling `new` on `require`.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The `require` function is used to include modules and might return a constructor. As this
```

Comment:
We should mention that `require` is a non-newable function, so if they're not using a custom function also called `require` then they'll get a runtime error.

Reply:
test.js
```javascript
const x = new require("./lol.js");
console.log(x);
```

foo.js
```javascript
class Foo {
};

module.exports = Foo;
```

This does not throw a runtime error with nodejs 22.

Reply:
Similar example here https://nodejs.org/docs/v20.17.0/api/modules.html#modules-commonjs-modules

Reply:
You make a good point. This is pathological, but valid